compiler/rustc_codegen_llvm/src/va_arg.rs

```diff
@@ -82,10 +82,10 @@ fn emit_ptr_va_arg(
     let (addr, addr_align) =
         emit_direct_ptr_va_arg(bx, list, llty, size, align.abi, slot_size, allow_higher_align);
     if indirect {
-        let tmp_ret = bx.load(addr, addr_align);
-        bx.load(tmp_ret, align.abi)
+        let tmp_ret = bx.load(llty, addr, addr_align);
+        bx.load(bx.cx.layout_of(target_ty).llvm_type(bx.cx), tmp_ret, align.abi)
```

Comment:
```suggestion
        bx.load(bx.layout_of(target_ty).llvm_type(bx.cx), tmp_ret, align.abi)
```

Should work, I believe.